### PR TITLE
Fix recruitment

### LINKED ISSUE #18279: Recruitment mask (customized number) did not work
# Bug
The customized numbers in the Recruitment module did not work

## Environment
- **Version**: Dolibarr 14.0.0

### PR DESCRIPTION
**Pull request by cfoellmann**
# FIX
- numbering for jobs
- advanced numbering for applications

CLOSE #18279

Question is:

can we fix the naming?
"Candidature" makes no sense!! It is just wrong. It is an application.

All constants and parameters are far too long all over the place. It is a pain to debug or extend this.

I am open to do the work and change this bad naming as discussed here #14753

### CODE REVIEW

**Comment by eldy** [comment]
Yes Candidature is not the best term. "Application" is the good one.
However, this term should never appears to end user. Only for dev.

You can make PR to change this but with very small changes and atomic. For example, changing only the name of table and having code still ok. Another one for changing only the name of class without changing table. In the order of your choice but only one PR for on reason (table, or class or constants, or ..., not a global PR for all renaming).

What about this PR ? Seems to be in draft status. Can i merge it ?

**Comment by cfoellmann** [comment]
@eldy Thanks for being open for the naming change.

This PR is still in the works. I want to fix "everything" that is currently broken in the module.
There are still a lot of bugs

File: htdocs/recruitment/class/recruitmentjobposition.class.php [hunk 1]
```diff
@@ -105,13 +105,13 @@ class RecruitmentJobPosition extends CommonObject
...
 		'fk_establishment' => array('type'=>'integer:Establishment:hrm/class/establishment.class.php', 'label'=>'Establishment', 'enabled'=>'$conf->hrm->enabled', 'position'=>56, 'notnull'=>0, 'visible'=>-1, 'foreignkey'=>'establishment.rowid',),
 		'fk_soc' => array('type'=>'integer:Societe:societe/class/societe.class.php:1:status=1 AND entity IN (__SHARED_ENTITIES__)', 'label'=>'WorkPlace', 'enabled'=>'$conf->societe->enabled', 'position'=>57, 'notnull'=>-1, 'visible'=>-1, 'css'=>'maxwidth500', 'index'=>1, 'help'=>"IfJobIsLocatedAtAPartner", 'picto'=>'company'),
 		'date_planned' => array('type'=>'date', 'label'=>'DateExpected', 'enabled'=>'1', 'position'=>60, 'notnull'=>0, 'visible'=>1,),
-		'remuneration_suggested' => array('type'=>'varchar(255)', 'label'=>'Remuneration', 'enabled'=>'1', 'position'=>62, 'notnull'=>0, 'visible'=>1,),
+		'remuneration_suggested' => array('type'=>'integer', 'label'=>'Remuneration', 'enabled'=>'1', 'position'=>62, 'notnull'=>0, 'visible'=>1,),
```
Comment (eldy) [inline htdocs/recruitment/class/recruitmentjobposition.class.php:114]: We must keep a free text here.
If may be a text like "Depends on ..." or "To negociate..."

Comment (cfoellmann) [reply htdocs/recruitment/class/recruitmentjobposition.class.php:114]: all other renumerations_ are int with char it is not comparable

Comment (cfoellmann) [reply htdocs/recruitment/class/recruitmentjobposition.class.php:114]: I will remove for now

**Pull request merged**